src/odemis/util/driver.py

```diff
@@ -157,12 +160,55 @@ def guessActuatorMoveDuration(actuator, axis, distance, accel=DEFAULT_ACCELERATI
     if not axis in actuator.axes:
         raise KeyError("The actuator component %s is expected to have %s axis, but it does not." % (actuator, axis))
     if model.hasVA(actuator, "speed"):
-        speed = actuator.speed.value.get("z", None)
+        speed = actuator.speed.value.get(axis, None)
     if speed is None:
         speed = DEFAULT_SPEED  
     return estimateMoveDuration(distance, speed, accel) 
 
 
+class ProgressiveMove(model.ProgressiveFuture):
+    def __init__(self, comp, pos):
+        self.name = comp.name
+        est_start = time.time() + 0.1
+
+        # calculate the time the mirror needs to move from the current position to the requested position
+        current_pos = comp.position.value
+        current_speed = comp.speed.value
+        axes_total_move_time = 0.0
+
+        for ax in pos.keys():
+            if ax in current_speed.keys():
+                axes_total_move_time += abs((current_pos[ax] - pos[ax]) / current_speed[ax])
+            else:
+                # use a safe speed default value (0.1 m/s) for this axis instead
+                axes_total_move_time += abs((current_pos[ax] - pos[ax]) / 0.1)
```

Comment:
Please use `guessActuatorMoveDuration(comp, ax, (current_pos[ax] - pos[ax])`. It will do all this for you.